UT3 Hellbender Overall Improvements

HellDragon-HK opened the pull request:
This is everything I've changed with the Hellbender thus far which includes
-Drawscale (0.95 like Alex said it needed to be unless I misunderstood in which case back to the turret size and corona placement issues)
-Turret Size & Collision Update https://drive.google.com/open?id=1UVMWr1TS6UR3CvhCeTNkxDUJLrUm0wFS 
-Sound Pack Update for missing explosion sound https://drive.google.com/open?id=1seZJkSMcgqS5aNgjNgqlj598RCad_DYV 
--Needs Scorpion sound update as well  https://drive.google.com/open?id=1l7cC58tLhf-nAps3Dzk00FttPPchNaDP
-How much the vehicle is thrown around by weapon fire
-Speed
-Handling
-Exit Positions closer to UT3
-Headlights adjustments for smaller size
-Headlight Projector adjustment
-Brakelight adjustment

GreatEmerald commented:
I guess the suspension offset change should be here too. Also, I need to look at the DrawScale (or someone else can test this).

HellDragon-HK commented:
Suspension fix is already here at line 457

GreatEmerald commented on Classes/UT3Hellbender.uc:
```diff
@@ -1,6 +1,7 @@
 /*
  * Copyright © 2008, 2014 GreatEmerald
  * Copyright © 2008-2009 Wormbo
+ * Copyright © 2017 HellDragon (still too strong a word but if you want it, here you go)
```
Again, no comments allowed in the copyright line.
While you're at it, you can also add `, 2017` to my line.

HellDragon-HK replied:
Sure, do you want me to try and catch that on any others too along with GP?

GreatEmerald replied:
No. The year indicates when the person contributed. I contributed the way to add the function here, hence the year should be updated, but nowhere else.

GreatEmerald replied:
Also, don't update it like that, it needs to say `2008, 2014, 2017`!

GreatEmerald commented on Classes/UT3Hellbender.uc:
```diff
@@ -57,7 +58,8 @@ function AltFire(optional float F) //This is to remove the horn each time you fi
 {                            //Make sure you don't hurt yourself with a combo
-    if (InstigatedBy != self)
+    if (InstigatedBy == self && ClassIsChildOf(DamageType, class'DamTypeSkyMine'))
+    return;
```
Put four more spaces here.

GreatEmerald commented on Classes/UT3Hellbender.uc:
```diff
@@ -415,33 +418,84 @@ defaultproperties
+    GroundSpeed=800.000000 //700
     SoundVolume=255
 
-    DrawScale=1.0
```
By default the `DrawScale` is not 1.0. Yet if you scaled it up in the UKX to the right size, then `DrawScale` should be set to 1.0, because that means "no further changes to scale needed in the code".

HellDragon-HK replied:
It might be a good idea to set that as default yes because some of the stock classes have a drawscale setting we don't want interferring right?

GreatEmerald replied:
Indeed. You could double-check all these to make sure they're all 1.0, but just setting it to 1.0 makes it certain.

GreatEmerald commented on Classes/UT3Hellbender.uc:
```diff
@@ -415,33 +418,84 @@ defaultproperties
+    EngineInertia=0.01
+    WheelInertia=0.01
+    ChassisTorqueScale=0.82 //0.7
+    WheelSuspensionOffset=5.0 //this one little value causes the Hellbender to sit lower like it should, unbeliveble we went crazy trying to solve it and all it needed was this
```
I suggest not making lines longer than ~100 characters, otherwise it looks cut off

HellDragon-HK replied:
I can remove the comment, just upon finding that one little key value that allowed fixing every ground vehicle was one of those kind of moments know what I mean

GreatEmerald replied:
Sure, it's fun to document things like that, just perhaps not use that much space for it.

HellDragon-HK commented:
What other changes here are still being needed?  I set scale to a forced 1.0, it's just been moved up to where I usually have it which on this one is line 330.

GreatEmerald commented:
My copyright line; you should not remove 2014 from it.

GreatEmerald commented:
OK, you're missing the changes to defaultproperties. If you check the logs you'll see a whole lot of `Accessed None`s with regards to the Hellbender weapons, that's because the array of weapons is longer due to the vanilla Hellbender having more of them. That's probably what's causing this weird behaviour with the number buttons as well. I'm not sure what works, though, we need to experiment a bit.

First of all, remove `PassengerWeapons(1)=()`. Then try adding `PassengerWeapons=()` above `PassengerWeapons(0)=(WeaponPawnClass=Class'UT3HellbenderRearGunPawn',WeaponBone="MainTurretYaw")`. If that doesn't fix the accessed nones and the strange button behaviour, then try replacing `PassengerWeapons(0)=(WeaponPawnClass=Class'UT3HellbenderRearGunPawn',WeaponBone="MainTurretYaw")` with `PassengerWeapons=(WeaponPawnClass=Class'UT3HellbenderRearGunPawn',WeaponBone="MainTurretYaw")` or `PassengerWeapons=(0=(WeaponPawnClass=Class'UT3HellbenderRearGunPawn',WeaponBone="MainTurretYaw"))`.

HellDragon-HK commented:
The first option there is what I already had and just missed it in the request, the 2nd and 3rd I can't get to compile

redundant data:     PassengerWeapons=(0=(WeaponPawnClass=Class'UT3HellbenderRearGunPawn',WeaponBone="MainTurretYaw"))


    MovementAnims(0) = "Idle"

    DriverWeapons(0)=(WeaponClass=Class'UT3HellbenderSideGun',WeaponBone="SecondaryTurretYaw")
    PassengerWeapons=()
    PassengerWeapons=(0=(WeaponPawnClass=Class'UT3HellbenderRearGunPawn',WeaponBone="MainTurretYaw"))
    
    FlagBone = Hood;

GreatEmerald commented:
Ah yes, it says it's redundant because of the `PassengerWeapons=()` line, remove that one while testing this.

HellDragon-HK commented:
Uh the Hellbender has a turret vacancy

GreatEmerald commented:
Eh?

HellDragon-HK commented:
I mean it has no rear turret now

GreatEmerald commented:
Huh, that's weird. So what happens with:
```
PassengerWeapons=(WeaponPawnClass=Class'UT3HellbenderRearGunPawn',WeaponBone="MainTurretYaw")
```
And with:
```
PassengerWeapons=((WeaponPawnClass=Class'UT3HellbenderRearGunPawn',WeaponBone="MainTurretYaw"))
```

HellDragon-HK commented:
PassengerWeapons=(WeaponPawnClass=Class'UT3HellbenderRearGunPawn',WeaponBone="MainTurretYaw")

Still vacant

PassengerWeapons=((WeaponPawnClass=Class'UT3HellbenderRearGunPawn',WeaponBone="MainTurretYaw"))

This fixed it and no number 3 key either now nor respawn on back turret

GreatEmerald commented:
Excellent! So *that* is a great example of how to get rid of a turret! Turns out that you need one pair of `()` to show that this is an array, and the other pair of `()` inside the first one to show that this is the one and only element inside it.

HellDragon-HK commented:
Added in the 'Great example of how to get rid of a turret' (without actually saying it is, do you want that added as well for future reference on what was done in case it's needed later?)

Added in the team corona lines

GreatEmerald commented:
Yea, move that comment over there. And also remove all the lines that you now have commented out, they are no longer needed.

HellDragon-HK commented:
Changed, is that how you wanted it?

GreatEmerald commented on Classes/UT3Hellbender.uc:
```diff
@@ -415,33 +302,80 @@ defaultproperties
+    ExplosionSounds(1) = Sound'UT3A_Vehicle_Hellbender.Sounds.A_Vehicle_Hellbender_Explode01';
+    ExplosionSounds(2) = Sound'UT3A_Vehicle_Hellbender.Sounds.A_Vehicle_Hellbender_Explode01';
+    ExplosionSounds(3) = Sound'UT3A_Vehicle_Hellbender.Sounds.A_Vehicle_Hellbender_Explode01';
+    ExplosionSounds(4) = Sound'UT3A_Vehicle_Hellbender.Sounds.A_Vehicle_Hellbender_Explode01';
```
All of these should also be folded into single arrays. Put `ExplosionSounds=()` and `ImpactDamageSounds=()` above and remove the duplicate entries.

GreatEmerald commented:
Oh, also, I hope that you're not seeing `Accessed None`s in the log with regards to the Hellbender any more now?

HellDragon-HK commented:
What do you mean by status of binaries?   Every binary file I've worked with to date is on G-Drive currently I'm fairly sure.

I'm not, that was something I noticed right away

GreatEmerald commented:
Uh, that's not quite how you are supposed to clear the arrays.

So all the binaries you have in `Binaries.csv` already refer to the scale at which the vehicles need to be for the headlight offsets etc. to be correct?

HellDragon-HK commented:
How do I need to do it then?

Should be yes

GreatEmerald commented:
I said `ExplosionSounds=()` and `ImpactDamageSounds=()`.

Yea, the manta's headlight in your screenshot looked weird, but that's for another issue I suppose.

GreatEmerald commented:
OK, thanks.
